pkg/loop/internal/pb/relayer.proto

```diff
@@ -0,0 +1,262 @@
+syntax = "proto3";
+
+option go_package = "github.com/smartcontractkit/chainlink-relay/pkg/loop";
+
+package loop;
+
+import "google/protobuf/empty.proto";
+
+service PluginRelayer {
+  rpc NewRelayer (NewRelayerRequest) returns (NewRelayerReply) {}
+}
+
+message NewRelayerRequest {
+  string config = 1; // toml (is chain instance config enough?)
+  uint32 keystoreID = 2;
+
+  //TODO prometheus? https://smartcontract-it.atlassian.net/browse/BCF-2075
+}
+
+message NewRelayerReply {
+  uint32 relayerID = 1;
+}
+
+service Keystore {
+  rpc Keys (google.protobuf.Empty) returns (KeysReply) {}
+  rpc Sign (SignRequest) returns (SignReply) {}
+}
+
+message KeysReply {
+  repeated string accounts = 1;
+}
+
+message SignRequest {
+  string account = 1;
+  bytes data = 2;
+}
+
+message SignReply {
+  bytes signed = 1;
+}
+
+service Relayer {
+  rpc NewConfigProvider (NewConfigProviderRequest) returns (NewConfigProviderReply) {}
+  rpc NewMedianProvider (NewMedianProviderRequest) returns (NewMedianProviderReply) {}
+  rpc NewMercuryProvider (NewMercuryProviderRequest) returns (NewMercuryProviderReply) {}
+
+  rpc ChainStatus (ChainStatusRequest) returns (ChainStatusReply) {}
+  rpc ChainStatuses (ChainStatusesRequest) returns (ChainStatusesReply) {}
+
+  rpc NodeStatuses (NodeStatusesRequest) returns (NodeStatusesReply) {}
+
+  rpc SendTx (SendTxRequest) returns (google.protobuf.Empty) {}
+}
+
+message RelayArgs {
+  bytes externalJobID = 1; // [32]byte
+  int32 jobID = 2;
+  string contractID = 3;
+  bool new = 4;
+  bytes relayConfig = 5;
+}
+
+message PluginArgs {
+  string transmitterID = 1;
+  bytes pluginConfig = 2;
+}
+
+message NewConfigProviderRequest {
+  RelayArgs relayArgs = 1;
+}
+
+message NewConfigProviderReply {
+  uint32 configProviderID = 1;
+}
+
+message NewMedianProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+message NewMedianProviderReply {
+  uint32 medianProviderID = 1;
+}
+
+message NewMercuryProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+message NewMercuryProviderReply {
+  uint32 mercuryProviderID = 1;
+}
+
+message ChainStatusRequest {
+  string id = 1;
+}
+
+message ChainStatusReply {
+  ChainStatus chain = 1;
+}
+
+message ChainStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+}
+
+message ChainStatusesReply {
+  repeated ChainStatus chains = 1;
+  int32 count = 2;
```

Comment:
Nearly all of this API is acting as a middle-ware for existing types, so I just started with the simplest version. I'm not opposed to changing these, but it would have cascading effects in legacy core code, so I'd rather do it separately.